restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -154,6 +159,128 @@ public ReactiveDocumentService(
+
+                // shred rows
+                List<JsonShreddedRow> rows =
+                    jsonDocumentShredder.shred(documentNode, Collections.emptyList());
+
+                // add to map and make sure we did not have already the same ID
+                if (documentRowsMap.put(documentId, rows) != null) {
+                  String msg =
+                      String.format(
+                          "Found duplicate ID %s in more than one document when doing batched document write.",
+                          documentId);
+                  throw new ErrorCodeRuntimeException(
+                      ErrorCode.DOCS_API_WRITE_BATCH_DUPLICATE_ID, msg);
+                }
+              }
+
+              // generate the Single for each document write
+              // the pair contains document id and boolean that denotes if write was successful
+              List<Single<Pair<String, Boolean>>> writeAll =
+                  documentRowsMap.entrySet().stream()
+                      .map(
+                          entry -> {
+                            String documentId = entry.getKey();
+                            List<JsonShreddedRow> documentRows = entry.getValue();
+
+                            // use write when possible to avoid the extra delete query
+                            return Single.defer(
+                                    () -> {
+                                      if (useUpdate) {
+                                        return writeService.updateDocument(
+                                            db.getQueryExecutor().getDataStore(),
+                                            namespace,
+                                            collection,
+                                            documentId,
+                                            documentRows,
+                                            db.treatBooleansAsNumeric(),
+                                            context);
+                                      } else {
+                                        return writeService.writeDocument(
+                                            db.getQueryExecutor().getDataStore(),
+                                            namespace,
+                                            collection,
+                                            documentId,
+                                            documentRows,
+                                            db.treatBooleansAsNumeric(),
+                                            context);
+                                      }
+                                    })
+
+                                // if successful return pair with true
+                                .map(result -> Pair.with(documentId, true))
+
+                                // on any error mark this document id as failed
+                                .onErrorReturn(t -> Pair.with(documentId, false));
```

Comment:
Is this the exception information thrown away here? Or is this handled exception logged somewhere else?

Reply:
Ok we can log the exception for sure..